haptools/sim_genotype.py

```diff
@@ -153,12 +167,19 @@ def output_vcf(
             # create ref_gts from mapping hap_samples to their respective genotypes 
             ref_sample_inds_chrom = np.repeat(hap_samples_ind, inter_len)
 
+            # select which alleles for each segment
+            if no_replacement:
+                ref_sample_haps_chrom = np.repeat(hap_sample_haps, inter_len)
+
+
             # grab random haplotype from samples and use as gts for our simulated samples
             end_var = cur_var+ref_sample_inds_chrom.shape[0]
             gt_vars = np.arange(cur_var, end_var, 1)
+            if not no_replacement:
+                ref_sample_haps_chrom = np.random.randint(2, size=len(gt_vars))
```

Comment:
Minor: might be clearer if this is an `else:` branch on the `if no_replacement:` code a few lines up.

Reply:
Actually, it seems like we want something like
`np.repeat(np.random.randint(2, size=len(inter_len)), inter_len)`
Otherwise (unless I'm misunderstanding what this does), you'll end up introducing switch errors into the haplotypes and they won't closely resemble real data.